flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/nodes/physical/stream/StreamExecLimit.scala

```diff
@@ -72,4 +89,78 @@ class StreamExecLimit(
       .item("fetch", RelExplainUtil.fetchToString(fetch))
   }
 
+  //~ ExecNode methods -----------------------------------------------------------
+
+  override def getInputNodes: util.List[ExecNode[StreamTableEnvironment, _]] = {
+    List(getInput.asInstanceOf[ExecNode[StreamTableEnvironment, _]])
+  }
+
+  override protected def translateToPlanInternal(
+      tableEnv: StreamTableEnvironment): StreamTransformation[BaseRow] = {
+    if (fetch == null) {
+      throw new TableException(
+        "FETCH is missed, which on streaming table is not supported currently")
+    }
+    val inputRowTypeInfo = FlinkTypeFactory.toInternalRowType(getInput.getRowType).toTypeInfo
+    val generateRetraction = StreamExecRetractionRules.isAccRetract(this)
+    val tableConfig = tableEnv.getConfig
+    val minIdleStateRetentionTime = tableConfig.getMinIdleStateRetentionTime
+    val maxIdleStateRetentionTime = tableConfig.getMaxIdleStateRetentionTime
+
+    // rankStart begin with 1
+    val rankRange = new ConstantRankRange(limitStart + 1, limitEnd)
+    val rankType = RankType.ROW_NUMBER
+    val outputRankNumber = false
+    // TODO Use RankFunction underlying StreamExecLimit currently
```

Comment:
Why we use RankFunction instead of a codegened sorter just like StreamExecSort did? Same question to StreamExecSortLimit

Reply:
Because SortLimit and Limit takes `limited` elements from `offset`, which is very similar to TopN, and could use AppendRankFunction or UpdateRankFunction under some as optimization in some scenarios. BTW, maybe RankFunction is not a good name, it's only for TopN.